WordPress/src/main/java/org/wordpress/android/ui/reader/usecases/ReaderSiteFollowUseCase.kt

```diff
@@ -28,44 +30,56 @@ class ReaderSiteFollowUseCase @Inject constructor(
 ) {
     private var continuation: Continuation<Boolean>? = null
 
-    suspend fun toggleFollow(post: ReaderPost) = flow<FollowSiteState> {
+    suspend fun toggleFollow(param: Param) = flow<FollowSiteState> {
         if (!networkUtilsWrapper.isNetworkAvailable()) {
             emit(NoNetwork)
         } else {
-            val isAskingToFollow = !readerPostTableWrapper.isPostFollowed(post)
-            val showEnableNotification = !readerUtilsWrapper.isExternalFeed(post.blogId, post.feedId) &&
-                    !post.isFollowedByCurrentUser
-            emit(PostFollowStatusChanged(post.blogId, isAskingToFollow, showEnableNotification))
-
-            performAction(post, isAskingToFollow)
+            val isAskingToFollow = when (param) {
+                is FromPost -> !readerPostTableWrapper.isPostFollowed(param.post)
+                is RecommendedSite -> param.isAskingToFollow
+            }
+            val showEnableNotification = when (param) {
+                is FromPost -> !readerUtilsWrapper.isExternalFeed(param.blogId, param.feedId) &&
+                        !param.post.isFollowedByCurrentUser
+                is RecommendedSite -> param.isAskingToFollow // todo: figure out logic for recommended site
```

Comment:
What should be the logic for the recommended site here?

Reply:
According to [Chris answer](https://appsp2.wordpress.com/2020/09/04/trial-task-sites-to-follow-discovery-card-thoughts-ideas/#comment-2386) this logic is fine for recommended blogs because API won't return external feeds. 

I think we could simplify it to sth like 
```kotlin
val showEnableNotification = !readerUtilsWrapper.isExternalFeed(param.blogId, param.feedId) && isAskingToFollow
```

WDYT?

Reply:
LGTM 👍